src/BusinessLayer/PacketReporting.cpp

```diff
@@ -0,0 +1,112 @@
+#include "BatteryData.h"
+#include "BatteryFaultsData.h"
+#include "DriverControlsData.h"
+#include "InternetView.h"
+#include "KeyMotorData.h"
+#include "LightsData.h"
+#include "MotorDetailsData.h"
+#include "MotorFaultsData.h"
+#include "MpptData.h"
+#include "PacketReporting.h"
+#include "../UILayer/PacketWindow.h"
+#include "../BusinessLayer/Packet/BatteryFaultsReporting.h"
+#include "../BusinessLayer/Packet/BatteryReporting.h"
+#include "../BusinessLayer/Packet/DriverControlsReporting.h"
+#include "../BusinessLayer/Packet/KeyMotorReporting.h"
+#include "../BusinessLayer/Packet/LightsReporting.h"
+#include "../BusinessLayer/Packet/MotorDetailsReporting.h"
+#include "../BusinessLayer/Packet/MotorFaultsReporting.h"
+#include "../BusinessLayer/Packet/MpptReporting.h"
+#include <QPushButton>
+
+PacketReporting::PacketReporting(KeyMotorData& keyMotorData,
+                                 MotorDetailsData& motor0DetailsData,
+                                 MotorDetailsData& motor1DetailsData,
+                                 DriverControlsData& driverControlsData,
+                                 MotorFaultsData& motorFaultsData,
+                                 BatteryFaultsData& batteryFaultsData,
+                                 BatteryData& batteryData,
+                                 MpptData& mpptData,
+                                 LightsData& lightsData,
+                                 PacketWindow& window)
+    : keyMotorData_(keyMotorData)
+    , motor0DetailsData_(motor0DetailsData)
+    , motor1DetailsData_(motor1DetailsData)
+    , driverControlsData_(driverControlsData)
+    , motorFaultsData_(motorFaultsData)
+    , batteryFaultsData_(batteryFaultsData)
+    , batteryData_(batteryData)
+    , mpptData_(mpptData)
+    , lightsData_(lightsData)
+    , window_(window)
+{
+    connect(&(window_.setButton()), SIGNAL(clicked()),
+                     this, SLOT(setAll()));
+}
+
+void PacketReporting::setBattery()
+{
+    BatteryReporting batteryReporting(batteryData_, window_.batteryTab());
```

Comment:
Creating a new reporting object every time we want to change the data might be a little bit slow, let's make each of those objects a private variable.